creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -347,129 +424,272 @@ class SnapbackSM {
-   * @param {string[]} param.unhealthyReplicasSet an array of endpoints of unhealthy replica set nodes
+   * @param {Set<string>} param.unhealthyReplicasSet a set of endpoints of unhealthy replica set nodes
    * @param {string[]} param.healthyNodes array of healthy Content Node endpoints used for selecting new replica set
+   * @param {Object} replicaSetNodesToUserClockStatusesMap map of secondary endpoint strings to (map of user wallet strings to clock value of secondary for user)
    * @returns {Object}
    * {
-   *  newPrimary: {string} the endpoint of the newly selected primary,
-   *  newSecondary1: {string} the endpoint of the newly selected secondary #1,
-   *  newSecondary2: {string} the endpoint of the newly selected secondary #2
+   *  newPrimary: {string | null} the endpoint of the newly selected primary or null,
+   *  newSecondary1: {string | null} the endpoint of the newly selected secondary #1,
+   *  newSecondary2: {string | null} the endpoint of the newly selected secondary #2,
+   *  issueReconfig: {boolean} flag to issue reconfig or not
    * }
    */
-  async determineNewReplicaSet ({ primary, secondary1, secondary2, wallet, unhealthyReplicasSet, healthyNodes }) {
-    let currentNodeHasExistingUserState = true
-    let i = 0
-    let newSecondary
-    while (currentNodeHasExistingUserState && i++ < MAX_SELECT_NEW_REPLICA_SET_ATTEMPTS) {
-      // Make sure current replica set will not be chosen as potential nodes for the new replica set
-      const healthyPotentialPeers = healthyNodes.filter(node => node !== primary && node !== secondary1 && node !== secondary2)
+  async determineNewReplicaSet ({ primary, secondary1, secondary2, wallet, unhealthyReplicasSet, healthyNodes, replicaSetNodesToUserClockStatusesMap }) {
+    let response = { newPrimary: null, newSecondary1: null, newSecondary2: null, issueReconfig: false }
+
+    const currentReplicaSet = [primary, secondary1, secondary2]
+    const healthyReplicaSet = new Set(currentReplicaSet.filter(node => !unhealthyReplicasSet.has(node)))
+    const newReplicaNodes = await this.selectRandomReplicaSetNodes({
+      healthyReplicaSet,
+      numberOfUnhealthyReplicas: unhealthyReplicasSet.size,
+      healthyNodes,
+      wallet
+    })
+
+    // Note: for v0, only issue reconfigs if only 1 secondary is unhealthy. Else, just log out the tentative
+    // new replica set to be assigned.
+
+    let newPrimary
+    if (unhealthyReplicasSet.size === 1) {
+      if (unhealthyReplicasSet.has(primary)) {
+        // If snapbackSM has already checked this primary and it failed the health check, select the higher clock
+        // value of the two secondaries as the new primary, leave the other as the first secondary, and select a new second secondary
+        let currentHealthySecondary
+        ([newPrimary, currentHealthySecondary] = replicaSetNodesToUserClockStatusesMap[secondary1][wallet] >= replicaSetNodesToUserClockStatusesMap[secondary2][wallet]
```

Comment:
seems like there is a scenario where a node-wallet pair could fail to exist in this map, we should handle that here?

Reply:
hm i actually don't think this case will occur. [here](https://github.com/AudiusProject/audius-protocol/pull/1600/files/56081646182e4f01112ece1bd5fc585f70ba46c3#diff-62e7d629775fe1c67406f64ab0a79b6aa6ed5bfb98fe5ed4e2048dc45bf4ed88R683-R686), we throw if from fetching a secondary clock value that it cannot be retrieved.

Reply:
^ my comment is no longer relevant as we now add a secondary to the unhealthyPeers set if a clock value cannot be retrieved. but actually, I don't think your original comment will happen either. 

fetching a clock value off `replicaSetNodesToUserClockStatusesMap` will be undefined if a secondary for a wallet did not return a clock value. in this scenario, the secondary will be marked as unhealthy (with my recent changes), and when a reconfig needs to be issued, will not go into this case as this case only pertains to only the primary being unhealthy

Reply:
wait why would a secondary be marked as unhealthy if it does not return a clock value for a wallet? that does not indicate unhealthy, simply requires a sync

Reply:
is this not what we discussed? upon clock check, if a secondary has not returned a proper clock value, add to unhealthyReplicaSet. https://github.com/AudiusProject/audius-protocol/pull/1600/files?file-filters%5B%5D=.js#diff-62e7d629775fe1c67406f64ab0a79b6aa6ed5bfb98fe5ed4e2048dc45bf4ed88R601-R604

Reply:
i think we just meant diff things. i meant that if it returns no data and you meant the web req failed. all good
also with latest clock values changes this scenario won't happen anyway, can resolve